### PR TITLE
Clean up SchedulerPredicates e2e tests

### PR DESCRIPTION
**Pull request by ivan4th**
Reduce repetition & don't use hardcoded pause image tags.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/33451)

### CODE REVIEW

**Review by gmarek — COMMENTED** [review]
One nit. Generally LGTM.

File: test/e2e/scheduler_predicates.go [hunk 1]
```diff
@@ -231,6 +98,29 @@ var _ = framework.KubeDescribe("SchedulerPredicates [Serial]", func() {
 
 	})
 
+	runAndKeepPodWithLabelAndGetNodeName := func() (string, string) {
```
Comment (ivan4th) [inline test/e2e/scheduler_predicates.go]: Fixed, made these toplevel.

**Comment by ivan4th** [comment]
@k8s-bot unit test this

Flake -- good ol' data race in federation controller test #32685

**Comment by gmarek** [comment]
LGTM - great work.

**Comment by ivan4th** [comment]
Thanks!

**Pull request merged**